src/main/java/hu/dungeonhunter/tools/Fight.java

```diff
@@ -79,6 +99,12 @@ public void championAttack() {
         } else {
             killedMonsterCounter++;
             monsterCounter--;
+            healingPotionCounter++;
+            System.out.println("You found a healing potion!");
+            if (healingPotionCounter > 5){
+                System.out.println("You can't have more than 5 healing potions!");
+                healingPotionCounter = 5;
+            }
```

Comment:
```suggestion
            System.out.println("You found a healing potion!");
            if (healingPotionCounter < 5) {
                healingPotionCounter++;
            } else {
                System.out.println("You can't have more than 5 healing potions!");
            }
```
You should only increase the counter if the counter is less than 5. This is a general rule, only do something if the conditions are met.
This is better, than increasing the counter, checking if it's greater than 5 and setting it back to 5 if it was greater.

Reply:
Ty, it really makes more sense that way. I rewrited.